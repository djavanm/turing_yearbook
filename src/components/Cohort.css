* {
  margin: 0;
  box-sizing: border-box;
}

.people-container {
  padding: 20px;
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-gap: 20px;
}

h2 {
  display: block;
}