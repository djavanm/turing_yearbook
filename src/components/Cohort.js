import React from 'react';
import Person from './Person';
import './Cohort.css';

const Cohort = () => {
  return (
    <div>
      Hi, I'm a person
    </div>
  )
};

export default Cohort;
