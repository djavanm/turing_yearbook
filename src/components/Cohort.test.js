import React from 'react';
import { shallow } from 'enzyme';
import Cohort from './Cohort';

const staff = [
{id: 1, name: 'Eric', quote: 'You miss all the shots you don\'t take.', superlative: 'Most Likely to Win the PGA', photo: './eric.jpg'},
{id: 2, name: 'Amy', quote: 'I believe in you, cuties!', superlative: 'Most Likely to Be an Actual Superhero', photo: './amy.jpg'},
{id: 3, name: 'Khalid', quote: 'Don\'t put the world on your shoulders.', superlative: 'Most Likely to Save the Planet from Alien Invasion', photo: './khalid.png'},
{id: 4, name: 'Travis', quote: 'You do you.', superlative: 'Most Likely to Win a Grammy', photo: './travis.jpg'},
{id: 5, name: 'Leta', quote: 'It\'s okay to be kind of a mess.', superlative: 'Most Likely to Go to the Moon', photo: './leta.jpg'},
{id: 6, name: 'Christie', quote: 'Every journey begins with a single step. So take the damn step!', superlative: 'Most Likely to Be President', photo: './christie.jpg'},
{id: 7, name: 'Brittany', quote: 'It was Robbie.', superlative: 'Most Likely to Get Away with Murder', photo: './brittany.jpg'},
{id: 8, name: 'Robbie', quote: 'Cool.', superlative: 'Most Likely to Be Accused of Murder', photo: './robbie.jpg'},
{id: 9, name: 'Will', quote: 'Stop worrying and just do the thing.', superlative: 'Most Likely to Invent Cold Fusion', photo: './will.jpg'},
{id: 10, name: 'Pam', quote: 'Shit!', superlative: 'Most Likely to Win a Gold Medal', photo: './pam.jpg'},
{id: 11, name: 'David', quote: 'I love git.', superlative: 'Most Likely to Be Eaten by a Shark', photo: './david.JPG'},
{id: 12, name: 'Louisa', quote: 'YOLO', superlative: 'Most Likely to Win a Knife Fight', photo: './louisa.jpg'},
{id: 13, name: 'Jeff', quote: '::thoughtful nonsense::', superlative: 'Most Likely to Tell a Dad Joke', photo: 'https://i.ytimg.com/vi/DhqzMc_LXgQ/maxresdefault.jpg'}
]

describe('Cohort', () => {
  it('matches the Cohort snapshot created for Form ', () => {
    const updatePersonMock = jest.fn();
    const removePersonMock = jest.fn();
    const wrapper = shallow(<Cohort people={staff} groupName='Staff' editable={false} updatePerson={updatePersonMock} removePerson={removePersonMock} />);

    expect(wrapper).toMatchSnapshot();
  });
});