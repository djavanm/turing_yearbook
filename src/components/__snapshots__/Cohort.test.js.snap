// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cohort matches the Cohort snapshot created for Form  1`] = `
<Fragment>
  <h2>
    { Staff }
  </h2>
  <section
    className="people-container"
  >
    <Person
      editable={false}
      key="1"
      person={
        Object {
          "id": 1,
          "name": "Eric",
          "photo": "./eric.jpg",
          "quote": "You miss all the shots you don't take.",
          "superlative": "Most Likely to Win the PGA",
        }
      }
      removePerson={[MockFunction]}
      updatePerson={[MockFunction]}
    />
    <Person
      editable={false}
      key="2"
      person={
        Object {
          "id": 2,
          "name": "Amy",
          "photo": "./amy.jpg",
          "quote": "I believe in you, cuties!",
          "superlative": "Most Likely to Be an Actual Superhero",
        }
      }
      removePerson={[MockFunction]}
      updatePerson={[MockFunction]}
    />
    <Person
      editable={false}
      key="3"
      person={
        Object {
          "id": 3,
          "name": "Khalid",
          "photo": "./khalid.png",
          "quote": "Don't put the world on your shoulders.",
          "superlative": "Most Likely to Save the Planet from Alien Invasion",
        }
      }
      removePerson={[MockFunction]}
      updatePerson={[MockFunction]}
    />
    <Person
      editable={false}
      key="4"
      person={
        Object {
          "id": 4,
          "name": "Travis",
          "photo": "./travis.jpg",
          "quote": "You do you.",
          "superlative": "Most Likely to Win a Grammy",
        }
      }
      removePerson={[MockFunction]}
      updatePerson={[MockFunction]}
    />
    <Person
      editable={false}
      key="5"
      person={
        Object {
          "id": 5,
          "name": "Leta",
          "photo": "./leta.jpg",
          "quote": "It's okay to be kind of a mess.",
          "superlative": "Most Likely to Go to the Moon",
        }
      }
      removePerson={[MockFunction]}
      updatePerson={[MockFunction]}
    />
    <Person
      editable={false}
      key="6"
      person={
        Object {
          "id": 6,
          "name": "Christie",
          "photo": "./christie.jpg",
          "quote": "Every journey begins with a single step. So take the damn step!",
          "superlative": "Most Likely to Be President",
        }
      }
      removePerson={[MockFunction]}
      updatePerson={[MockFunction]}
    />
    <Person
      editable={false}
      key="7"
      person={
        Object {
          "id": 7,
          "name": "Brittany",
          "photo": "./brittany.jpg",
          "quote": "It was Robbie.",
          "superlative": "Most Likely to Get Away with Murder",
        }
      }
      removePerson={[MockFunction]}
      updatePerson={[MockFunction]}
    />
    <Person
      editable={false}
      key="8"
      person={
        Object {
          "id": 8,
          "name": "Robbie",
          "photo": "./robbie.jpg",
          "quote": "Cool.",
          "superlative": "Most Likely to Be Accused of Murder",
        }
      }
      removePerson={[MockFunction]}
      updatePerson={[MockFunction]}
    />
    <Person
      editable={false}
      key="9"
      person={
        Object {
          "id": 9,
          "name": "Will",
          "photo": "./will.jpg",
          "quote": "Stop worrying and just do the thing.",
          "superlative": "Most Likely to Invent Cold Fusion",
        }
      }
      removePerson={[MockFunction]}
      updatePerson={[MockFunction]}
    />
    <Person
      editable={false}
      key="10"
      person={
        Object {
          "id": 10,
          "name": "Pam",
          "photo": "./pam.jpg",
          "quote": "Shit!",
          "superlative": "Most Likely to Win a Gold Medal",
        }
      }
      removePerson={[MockFunction]}
      updatePerson={[MockFunction]}
    />
    <Person
      editable={false}
      key="11"
      person={
        Object {
          "id": 11,
          "name": "David",
          "photo": "./david.JPG",
          "quote": "I love git.",
          "superlative": "Most Likely to Be Eaten by a Shark",
        }
      }
      removePerson={[MockFunction]}
      updatePerson={[MockFunction]}
    />
    <Person
      editable={false}
      key="12"
      person={
        Object {
          "id": 12,
          "name": "Louisa",
          "photo": "./louisa.jpg",
          "quote": "YOLO",
          "superlative": "Most Likely to Win a Knife Fight",
        }
      }
      removePerson={[MockFunction]}
      updatePerson={[MockFunction]}
    />
    <Person
      editable={false}
      key="13"
      person={
        Object {
          "id": 13,
          "name": "Jeff",
          "photo": "https://i.ytimg.com/vi/DhqzMc_LXgQ/maxresdefault.jpg",
          "quote": "::thoughtful nonsense::",
          "superlative": "Most Likely to Tell a Dad Joke",
        }
      }
      removePerson={[MockFunction]}
      updatePerson={[MockFunction]}
    />
  </section>
</Fragment>
`;
